cpp/include/cudf/ast/expressions.hpp

```diff
@@ -552,6 +557,86 @@ class column_name_reference : public expression {
   std::string column_name;
 };
 
+/**
+ * @brief An AST expression tree. it owns and contains multiple dependent expressions. All the
+ * expressions are destroyed once the tree is destructed.
+ */
+class tree {
+ public:
+  /**
+   * @brief construct an empty ast tree
+   */
+  tree()                  = default;
+  tree(tree&&)            = default;
+  tree& operator=(tree&&) = default;
+  ~tree()                 = default;
+
+  // the tree is not copyable
+  tree(tree const&)            = delete;
+  tree& operator=(tree const&) = delete;
+
+  /**
+  @brief Add an expression to the AST tree
+  @param args Arguments to use to construct the ast expression
+  @returns a reference to the added expression
+ */
```

Comment:
*nit*: throughout, these comments are weirdly formatted?

```suggestion
  /**
   * @brief Add an expression to the AST tree
   *
   * @param args Arguments to use to construct the ast expression
   * @returns a reference to the added expression
   */
```

Reply:
thanks for spotting this